apps/src/lib/ui/SchoolInfoInterstitial.jsx

```diff
@@ -0,0 +1,206 @@
+import React, {PropTypes} from 'react';
+import $ from 'jquery';
+import i18n from '@cdo/locale';
+import color from '../../util/color';
+import BaseDialog from '../../templates/BaseDialog';
+import Button from '../../templates/Button';
+import SchoolInfoInputs, {SCHOOL_TYPES_HAVING_NCES_SEARCH} from '../../templates/SchoolInfoInputs';
+
+const styles = {
+  container: {
+    margin: 20,
+    color: color.charcoal
+  },
+  heading: {
+    fontSize: 16,
+    fontFamily: "'Gotham 5r', sans-serif",
+  },
+  middle: {
+    marginTop: 20,
+    marginBottom: 20,
+    paddingBottom: 20,
+    paddingTop: 20,
+    borderTopWidth: 1,
+    borderBottomWidth: 1,
+    borderRightWidth: 0,
+    borderLeftWidth: 0,
+    borderStyle: 'solid',
+    borderColor: color.lighter_gray,
+  },
+  bottom: {
+    display: 'flex',
+    justifyContent: 'flex-end',
+  },
+  error: {
+    color: color.red,
+  },
+};
+
+export default class SchoolInfoInterstitial extends React.Component {
+  static propTypes = {
+    // This component is tightly bound to the HAML view that renders it and
+    // populates its props, and similarly to the User update API that
+    // it uses to save entered school info.
+    scriptData: PropTypes.shape({
+      formUrl: PropTypes.string.isRequired,
+      authTokenName: PropTypes.string.isRequired,
+      authTokenValue: PropTypes.string.isRequired,
+      existingSchoolInfo: PropTypes.shape({
+        // Note, these names intentionally match the fields on the SchoolInfo
+        // model, not their JavaScript-friendly equivalents.  The mapping
+        // occurs in the constructor, below, and back when we submit.
+        school_id: PropTypes.string,
+        country: PropTypes.string,
+        school_type: PropTypes.string,
+        school_name: PropTypes.string,
+        state: PropTypes.string,
+        zip: PropTypes.string,
+        full_address: PropTypes.string,
+      }).isRequired,
+    }).isRequired,
+    onClose: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    onClose: function () {},
```

Comment:
Does it actually make sense to use this component without providing a meaningful onClose?

Reply:
Only in a storybook/test context.  I'll clear this out.